### PR TITLE
Ensure strings as binaries

### PR DESCRIPTION
**Pull request by lukebakken**
Includes commits from #271 (CLIENTS-363) 

Depends on:

* basho/riak_kv#1388
* basho/riak_pb#185

### CODE REVIEW

File: src/riakc_ts_query_operator.erl [hunk 1]
```diff
@@ -48,7 +47,12 @@ serialize_interpolations([{Key, Value} | RemainingInterps],
     UpdatedInterps = [#rpbpair{key=Key, value=Value} | SerializedInterps],
     serialize_interpolations(RemainingInterps, UpdatedInterps).
 
-deserialize({error, Message}) -> {error, Message};
+deserialize({error, {Code, Message}}) when is_integer(Code), is_list(Message) ->
```
Comment (hmmr) [inline src/riakc_ts_query_operator.erl]: This is not necessary, for the reason that incoming messages don't ever contain strings (if the server sends any strings, the encoder will blow up somewhere and the message will not be delivered).

Comment (javajolt) [inline src/riakc_ts_query_operator.erl]: I was seeing strings in some tests.  Once I'm awake, I'll find some examples.

File: src/riakc_ts.erl [hunk 2]
```diff
@@ -186,3 +188,12 @@ server_call(Pid, Message) ->
     gen_server:call(Pid,
                     {req, Message, riakc_pb_socket:default_timeout(timeseries)},
                     infinity).
+-spec convert_to_binary(Value :: binary() | list()) -> binary().
```
Comment (javajolt) [inline src/riakc_ts.erl]: @lukebakken I thought the builtin `iolist_to_binary` was more elegant, which is why I didn't push my other PR.  Why switch to my home-brewed function?

File: src/riakc_ts.erl [hunk 3]
```diff
@@ -69,20 +71,20 @@ query(Pid, Query, Interpolations, Cover) when is_binary(Cover) ->
...
--spec get_coverage(pid(), table_name(), Query::string()) ->
-                          {ok, Entries::[term()]} | {error, term()}.
 %% @doc Generate a parallel coverage plan for the specified query
-get_coverage(Pid, Table, Query)
-  when is_pid(Pid), (is_binary(Table) orelse is_list(Table)), is_list(Query) ->
+-spec get_coverage(Pid::pid(),
+                   Table::binary(),
+                   QueryText::binary()) -> {ok, Entries::[term()]} | {error, term()} | {'EXIT', any()}.
+get_coverage(Pid, Table, QueryText) ->
     server_call(Pid,
```
Comment (hmmr) [inline src/riakc_ts.erl]: The Entries happen to be returned to the client raw. This means the caller of `riakc_ts:get_coverage` needs to do the matching on `#tscoverageentry{}` fields, while the record itself is internal to riak_pb. (Leaving this note here as a todo item, even though it goes outside of the scope of this PR).

File: src/riakc_ts.erl (hunk 2 above)
Comment (lukebakken) [inline src/riakc_ts.erl]: (shrug) I'll switch it back.

**Comment by javajolt** [comment]
I've got a couple of amendments coming.  Please don't merge quite yet.

File: src/riakc_ts_put_operator.erl [hunk 4]
```diff
@@ -31,12 +31,18 @@
...
     ColumnDescs = riak_pb_ts_codec:encode_columnnames(ColumnNames),
-    SerializedRows = riak_ttb_codec:encode_ts_rows(Measurements),
-    #tsputreq{table   = TableName,
+    #tsputreq{table   = iolist_to_binary(TableName),
               columns = ColumnDescs,
-              rows    = SerializedRows}.
+              rows    = Measurements}.
 
+deserialize({error, {Code, Message}}) when is_integer(Code), is_list(Message) ->
+    {error, {Code, iolist_to_binary(Message)}};
```
Comment (hmmr) [inline src/riakc_ts_put_operator.erl]: On the way back from the server, it's all binaries already -- no need to re-convert it again.

File: src/riakc_ts.erl [hunk 5]
```diff
@@ -122,10 +122,10 @@ put(Pid, Table, ColumnNames, Measurements)
 delete(Pid, Table, Key, Options)
   when is_pid(Pid), (is_binary(Table) orelse is_list(Table)),
        is_list(Key), is_list(Options) ->
-    Message = #tsdelreq{table   = iolist_to_binary(Table),
-                        key     = riak_pb_ts_codec:encode_cells_non_strict(Key),
-                        vclock  = proplists:get_value(vclock, Options),
-                        timeout = proplists:get_value(timeout, Options)},
+    Message = #tsdelreq{table  =iolist_to_binary(Table),
```
Comment (hmmr) [inline src/riakc_ts.erl]: These edits are mildly distracting and, well, go across the grain. Spaces around '=' are pretty much everywhere, and have won the riak world ages ago. All attempts to overthrow this style are futile.

File: src/riakc_ts_put_operator.erl (hunk 4 above)
Comment (javajolt) [inline src/riakc_ts_put_operator.erl]: Not true.  I was getting some error messages as strings.  There's been so much churn recently that might have changed.

File: src/riakc_ts.erl (hunk 5 above)
Comment (javajolt) [inline src/riakc_ts.erl]: I'm afraid I agree with @hmmr

File: src/riakc_ts_put_operator.erl (hunk 4 above)
Comment (hmmr) [inline src/riakc_ts_put_operator.erl]: Still, those need to be fixed first thing. All of them are in rpberrresp, and they are not many. Adding `iolist_to_binary` to riak_kv_pb_svc:make_errorresp would be enough.

File: src/riakc_ts.erl (hunk 5 above)
Comment (lukebakken) [inline src/riakc_ts.erl]: OK, I've seen about 50/50 use, but what's a few extra bytes.

**Comment by javajolt** [comment]
👍 c3f4b1f

**Pull request merged**